NetworkTest.java

```diff
@@ -0,0 +1,22 @@
+
+public class NetworkTest{
+    
+    public void networkTest(NetworkInterface interface)
+    {
+        //set the source for the input
+        InputResponse inputResponse = interface.inputSource(new InputSourceRequest());

```

Comment:
The checkstyle failure is from this not compiling - now that we've covered mock objects, that's the best way to fix it, but another option (that you may or may not have run into in CS1/2) is to create an anonymous inner class. It's similar to a lambda, but because there aren't any methods on the interface yet, you end up with the old pre-java-8 syntax: inputSource(new InputSourceRequest() {})